cdap-docs/developer-manual/source/metadata/programmatic-metadata.rst

```diff
@@ -0,0 +1,91 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2018 Cask Data, Inc.
+
+.. _programmatic-metadata:
+
+==================================
+Metadata from Programs and Plugins
+==================================
+
+.. highlight:: java
+
+Metadata can be accessed from *Programs* and *Plugins*. Currently, reading metadata from a program or plugin is not
+supported while it is running in *cloud* but metadata can be added, updated or deleted.
+
+.. _metadata-programs:
+
+Programs
+========
+
+Metadata can be accessed from *MapReduce*, *Spark*, *Workers* and *Service* through
+`MetadataReader <../../reference-manual/javadocs/co/cask/cdap/api/metadata/MetadataReader.html>`__ exposed to
+individual program context available in the ``initialize`` method.
+
+For example metadata can be read in *MapReduce* programs in the following way::
+
+  @Override
+  public void initialize() throws Exception {
+    MapReduceContext context = getContext();
+
+    // construct the metadata entity
+    MetadataEntity entity = MetadataEntity.builder().append(MetadataEntity.NAMESPACE, "myNamespace")
+      .appendAsType(MetadataEntity.DATASET, "myDataset").build();
+
+    // get the metadata
+    Map<MetadataScope, Metadata> metadata = context.getMetadata(entity);
+  }
+
+A program can also annotate metadata to an entity using the
+`MetadataWriter <../../reference-manual/javadocs/co/cask/cdap/api/metadata/MetadataWriter.html>`__
+exposed to same context object. For example a dataset can be tagged from a *MapReduce* program in the following way::
```

Comment:
for consistency with above, let's add a line break after the period?

Reply:
also change the sentence similar to the previous context?

Reply:
Done.